### PR TITLE
Percentages methods created for iteration 2

### PR DESCRIPTION
**Pull request by Mariana-21**
Created methods to get all percentages of home wins, away wins, and tied games. 
Was able to use a stub in the tied percentages, due to there not being a tied game in the CSV data.

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -69,4 +70,19 @@ def average_goals_by_season
       goals_by_season
     end
   end
+
+  def percentage_home_wins
+    home_wins = @game_collection.games.find_all {|game| game.home_goals > game.away_goals}
+    home_wins.count.to_f / (@game_collection.games.count.to_f).round(2)
```
Comment (cmpprg) [inline lib/stat_tracker.rb]: There is an all method that we should refactor with sometime in the future

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -69,4 +70,19 @@ def average_goals_by_season
...
+  end
+
+  def percentage_visitor_wins
+    home_wins = @game_collection.games.find_all {|game| game.home_goals < game.away_goals}
```
Comment (cmpprg) [inline lib/stat_tracker.rb]: The variable name here seems out of place, are we intending to say away_wins?

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -69,4 +70,19 @@ def average_goals_by_season
...
+    home_wins.count.to_f / (@game_collection.games.count.to_f).round(2)
+  end
+
+  def percentage_ties
+    tied_games = @game_collection.games.find_all {|game| game.home_goals == game.away_goals}
+    tied_games.count.to_f / (@game_collection.games.count.to_f).round(2)
```
Comment (cmpprg) [inline lib/stat_tracker.rb]: I think we should be using #length or @size for these, count is an enumerable and will soak up processing power.

File: test/stat_tracker_test.rb [hunk 4]
```diff
@@ -48,4 +50,18 @@ def test_it_can_show_average_goals_by_season
...
+  end
+
+  def test_it_can_get_away_win_percentages
+    assert_equal 0.3, @stat_tracker.percentage_visitor_wins
+  end
+
+  def test_it_can_get_tied_percentage
+    @stat_tracker.stubs(:percentage_ties).returns(0.10)
+
+    assert 0.10, @stat_tracker.percentage_ties
```
Comment (cmpprg) [inline test/stat_tracker_test.rb:73]: we can't use a stub here unfortunately. We are only testing that a stub returns the value that we want and not actually testing the method at all. we should all discuss modifying the truncated csv files. I am also running into some inadequacies in the data pool we are pulling from.

**Review by cmpprg — CHANGES_REQUESTED** [review]
-There are a few line comments needing review
-It seems to me that all the logic in the methods are solid.
-I am requesting a variable name change. 
-We should get together as a team to change some of the data in our truncated files so that we have more variety of outcomes, currently I am also running into a problem with the best_offense method because two teams meat the criteria.

File: lib/stat_tracker.rb (hunk 3 above)
Comment (Mariana-21) [reply lib/stat_tracker.rb]: Yes, I was thinking about using #count, I will change that now.

File: lib/stat_tracker.rb (hunk 1 above)
Comment (Mariana-21) [reply lib/stat_tracker.rb]: Yeah, I agree but I think we can do then when we get together and refactor it all together.

Comment (cmpprg) [reply lib/stat_tracker.rb]: sounds good.

**Comment by cmpprg** [comment]
Thanks for getting those changes taken care of.

**Pull request merged**